### ISSUE TITLE
Videos seem to look washed out

### ISSUE THREAD

**Issue by meichthys** [pulsejet/memories#406]
The trans-coded(?) videos seem to have a 'washed out' color to them, but in the iOS camera roll they look mostly the same. It's like the video's brightness was somehow turned up a few extra percent.

Video:
![image](https://user-images.githubusercontent.com/10717998/218239780-ca787d29-4a66-4dfe-b069-f5c777923ed8.png)

Photo:
![image](https://user-images.githubusercontent.com/10717998/218239810-f8b94dab-d70f-4569-9dfb-f0265b0824a9.png)

**Comment by vitis586** [comment]
Is it possible that the video is HDR? I always wondered how does ffmpeg handle those. I can imagine that if HDR is not accounted for the metadata can be lost and the video end up with different colourspace.

**Comment by meichthys** [comment]
@vitis586 yes, that could explain it for sure. The photo and video are both HDR: ![image](https://user-images.githubusercontent.com/10717998/218258351-e01b1eca-3aaa-49d6-8183-9e931b2e14d4.jpeg)

**Comment by meichthys** [comment]
For reference I can pretty much recreate the washed out look in iOS photos if I modify the video to have +50 brightness and -50 saturation:
![image](https://user-images.githubusercontent.com/10717998/218259328-b8ce3c12-ea0a-4bfa-a64a-9030878c4f06.png)

Original: ![image](https://user-images.githubusercontent.com/10717998/218259354-aee127dc-ebab-4b72-beff-ee15c7b68692.png)

**Comment by relink2013** [comment]
Figured id throw my vote in on this. I just recently upgraded from an iPhone SE to an iPhone 14 Pro and immediately noticed the same thing with all the videos from my new phone.

**Comment by Gymnae** [comment]
Yes, it's a problem for HDR videos, at least for those shot with iPhones. The generated preview image as well as transcoded video appear washed out. They are fine when played in "direct" mode.

**Comment by relink2013** [comment]
> I was able to fix it by using a different ffmpeg binary, the one supplied from jellyfin-ffmpeg fixes the washed out colors while transcoding

Quick question since we seem to have had similar ideas. Instead of copying the ffmpeg binary from Jellyfin, I actually just installed go-vod inside the Jellyfin container. It's working great but my videos are still washed out. Any tips?

The feature were all looking for is called HDR Tone Mapping, since no one actually said it, I figured I'd throw that out there in case it helps anyone.

**Comment by pulsejet** [comment]
The external go-vod container now uses jellyfin-ffmpeg, so this might be resolved

**Comment by gymnae** [comment]
Couldn’t the internal also ship with it as an portable binary?

**Comment by pulsejet** [comment]
Don't think such a thing exists. Afaik the best we can do is the debian packages of jellyfin-ffmpeg, but then these won't work on alpine (Linuxserver images). One possible alternative might be a custom dockerfile template to install jellyfin-ffmpeg using the deb packages in the Nextcloud container.

**Comment by pulsejet** [comment]
Earlier today, you'll need to pull the new image.
I don't think it works out of the box anyway, probably some extra options are needed. Can you provide a sample image video pair to test this?

**Comment by gymnae** [comment]
> Don't think such a thing exists. Afaik the best we can do is the debian packages of jellyfin-ffmpeg, but then these won't work on alpine (Linuxserver images). One possible alternative might be a custom dockerfile template to install jellyfin-ffmpeg using the deb packages in the Nextcloud container.

They do offer portable binaries which, afaik, work on Alpine: 
![image](https://github.com/pulsejet/memories/assets/6195487/1f10bce1-46b7-4eb5-a597-473ed3c16905)

In addition, I added custom jellyfin to my nextcloud docker image:
https://github.com/gymnae/owncloud/blob/master/Dockerfile

**Comment by pulsejet** [comment]
It's worth investigating, but I don't have high hopes.

> They do offer portable binaries which, afaik, work on Alpine

Did you test this? If this is the case then I'm very curious how they make it work. Does ffmpeg not depend on libc?

> In addition, I added custom jellyfin to my nextcloud docker image

Yes, I expect the debian packages to work fine, at least for sw transcoding. I'm just wary of shipping an ffmpeg binary _with_ memories; we could always add docs on how to install jellyfin-ffmpeg in the container / host.

**Comment by dronnikovigor** [comment]
> we could always add docs on how to install jellyfin-ffmpeg in the container

Hi, can you plz provide tutorial how to correctly build go-vod with jellyfin-ffmpeg?
